core/src/main/scala/org/apache/spark/memory/ExecutionMemoryPool.scala

```diff
@@ -151,7 +151,7 @@ private[memory] class ExecutionMemoryPool(
    */
   def releaseMemory(numBytes: Long, taskAttemptId: Long): Unit = lock.synchronized {
     val curMem = memoryForTask.getOrElse(taskAttemptId, 0L)
-    var memoryToFree = if (curMem < numBytes) {
+    val memoryToFree = if (curMem < numBytes) {
```

Comment:
Let's not put the different things in the same PR.

Reply:
@dongjoon-hyun I don't so much mind it if this is a comprehensive pass at a few different types of code cleanup

Reply:
I changed the title and description. @dongjoon-hyun please let me know if @srowen's suggenstion is OK with you.

Reply:
Yep. Please follow @srowen 's advice~